duckduckgo/macos-browser
# fix Tab reloadIfNeeded Task leaking Tab

Task/Issue URL: https://app.asana.com/0/1177771139624306/1204153833211831/f
Tech Design URL:
CC:

**Description**:
- Fixes regression Tab leakage in RELEASE builds (caused by WKNavigation objects not released instantly in RELEASE leading to async Task bound to Navigation lifetime keeps Tab)
- Fixed non-main-thread Tab deallocation issue in Unit Tests

**Steps to test this PR**:
1. Build for RELEASE
2. Make sure Always open in Duck Player setting is active
3. Search for youtube video, open in Duck Player
4. Close window, make sure video is stopped



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
